monkey/tests/unit_tests/infection_monkey/island_api_client/test_http_island_api_client.py

```diff
@@ -466,3 +467,40 @@ def test_request_retries(monkeypatch, island_api_client):
         island_api_client.connect(SERVER)
 
     assert mock_send.call_count == RETRIES + 1
+
+
+@pytest.mark.parametrize(
+    "status_code, expected_error",
+    [
+        (401, IslandAPIRequestError),
+        (501, IslandAPIRequestFailedError),
+    ],
+)
+def test_island_api_client_send_heartbeat__status_code(
+    island_api_client, status_code, expected_error
+):
+    with requests_mock.Mocker() as m:
+        m.get(ISLAND_URI)
+        island_api_client.connect(SERVER)
+
+        with pytest.raises(expected_error):
+            m.post(ISLAND_SEND_HEARTBEAT_URI, status_code=status_code)
+            island_api_client.send_heartbeat(agent_id=AGENT_ID, timestamp=TIMESTAMP)
+
+
+@pytest.mark.parametrize(
+    "actual_error, expected_error",
+    [
+        (requests.exceptions.ConnectionError, IslandAPIConnectionError),
+        (TimeoutError, IslandAPITimeoutError),
+        (Exception, IslandAPIError),
+    ],
+)
+def test_island_api_client_send_heartbeat__errors(island_api_client, actual_error, expected_error):
+    with requests_mock.Mocker() as m:
+        m.get(ISLAND_URI)
+        island_api_client.connect(SERVER)
+
+        with pytest.raises(expected_error):
+            m.post(ISLAND_SEND_HEARTBEAT_URI, exc=actual_error)
+            island_api_client.send_heartbeat(agent_id=AGENT_ID, timestamp=TIMESTAMP)
```

Comment:
We should remove all of these tests and add one test of each kind to another file since all the requests go through `infection_monkey/island_api_client/http_client.py`. There's no need for us to test each one separately here.

Reply:
We should use DI to inject the `HTTPClient` into `HTTPIslandAPIClient`'s constructor. If we do that, we'll need very few (if any) tests for `HTTPIslandAPIClient`.